api/src/opentrons/config/__init__.py

```diff
@@ -269,7 +269,8 @@ def _load_with_overrides(base) -> Dict[str, str]:
     should_write = False
     overrides = _get_environ_overrides()
     try:
-        index = json.load((base / _CONFIG_FILENAME).open())
+        with (base / _CONFIG_FILENAME).open() as file:
+            index = json.load(file)
```

Comment:
Since `(base / _CONFIG_FILENAME)` is a `pathlib.Path`, you can instead do this if you feel like it:

```suggestion
        index = json.loads((base / _CONFIG_FILENAME).read_text())
```

Ditto for the other places you use `with foo.open() as file`.

The difference, I suppose, is whether the entire file is slurped up into memory before handing it off to `json`. Our configuration files are small so I doubt it matters.